components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1122,6 +1126,7 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         return this.eligibilityService.isStudent(user);
     }
 
+    // **deprecated**
```

Comment:
Is this really deprecated?

To me, the BillingModes are only there to help with the transition to Stripe, but I thought that, as soon as all customers are migrated to Stripe, we'd delete the BillingModes and fall back to `getShowPaymentUI` again?

A very simple, boolean switch to turn off any payment UI seems like a must have, while replacing it with a somewhat complicated state machine with possible new bugs sounds riskier.

Reply:
Yes, that's debateable. 

> while replacing it with a somewhat complicated state machine with possible new bugs sounds riskier.

That's not what happens: When we got rid of Chargebee, it's a enum with two cases: "none" (which maybe should be named "license") and "usage-based". I think this is superior, because then we have a clear indicator as to what to show on the UI, for instance to deciding between showing self-hosted or SaaS concerns in the Admin dashboard.

But let's leave it as out-of scope for this PR.

Reply:
cmp [comment](https://github.com/gitpod-io/gitpod/pull/11814#issuecomment-1207714530).